reactor-netty-core/src/test/java/reactor/netty/FutureMonoTest.java

```diff
@@ -76,4 +89,132 @@ void testDeferredFutureMonoLater() {
+			final TestSubscriber subscriber = new TestSubscriber();
+			final ImmediateEventExecutor eventExecutor = ImmediateEventExecutor.INSTANCE;
+			final Promise<Void> promise = eventExecutor.newPromise();
+			final Supplier<Promise<Void>> promiseSupplier = () -> promise;
+
+			RaceTestUtils.race(() -> FutureMono.deferFuture(promiseSupplier)
+			                                   .subscribe(subscriber),
+					subscriber::cancel,
+					() -> promise.setSuccess(null));
+
+			assertThat(resolveListeners(promise)).isNullOrEmpty();
+			assertThat(subscriber.operations).first()
+			                                 .isEqualTo(TestSubscriber.Operation.ON_SUBSCRIBE);
+		}
+	}
+
+	@Test
+	void raceTestImmediateFutureMono() {
+		for (int i = 0; i < 1000; i++) {
+			final TestSubscriber subscriber = new TestSubscriber();
+			final ImmediateEventExecutor eventExecutor = ImmediateEventExecutor.INSTANCE;
+			final Promise<Void> promise = eventExecutor.newPromise();
+
+			RaceTestUtils.race(() -> FutureMono.from(promise)
+			                                   .subscribe(subscriber), subscriber::cancel);
+
+			assertThat(resolveListeners(promise)).isNullOrEmpty();
+			assertThat(subscriber.operations).first()
+			                                 .isEqualTo(TestSubscriber.Operation.ON_SUBSCRIBE);
+		}
+	}
+
+	@Test
+	void raceTestDeferredFutureMono() {
+		for (int i = 0; i < 1000; i++) {
+			final TestSubscriber subscriber = new TestSubscriber();
+			final ImmediateEventExecutor eventExecutor = ImmediateEventExecutor.INSTANCE;
+			final Promise<Void> promise = eventExecutor.newPromise();
+			final Supplier<Promise<Void>> promiseSupplier = () -> promise;
+
+			RaceTestUtils.race(() -> FutureMono.deferFuture(promiseSupplier)
+			                                   .subscribe(subscriber), subscriber::cancel);
+
+			assertThat(resolveListeners(promise)).isNullOrEmpty();
+			assertThat(subscriber.operations).first()
+			                                 .isEqualTo(TestSubscriber.Operation.ON_SUBSCRIBE);
+		}
+	}
+
+	static GenericFutureListener<? extends Future<?>>[] resolveListeners(Promise<Void> promise) {
+		try {
+			final Field listeners = DefaultPromise.class.getDeclaredField("listeners");
+			final Class<?> aClass = Class.forName("io.netty.util.concurrent.DefaultFutureListeners");
+			final Field listeners1 = aClass.getDeclaredField("listeners");
+
+			listeners.setAccessible(true);
+			listeners1.setAccessible(true);
+
+			final Object objListener = listeners.get(promise);
+			if (objListener == null) {
+				return null;
+			}
+			return (GenericFutureListener<? extends Future<?>>[]) listeners1.get(objListener);
+		}
+		catch (NoSuchFieldException | ClassNotFoundException | IllegalAccessException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	static class TestSubscriber implements CoreSubscriber<Void> {
+
+		enum Operation {
+			ON_ERROR, ON_COMPLETE, ON_SUBSCRIBE
+		}
+
+		volatile     Subscription                                              s;
```

Comment:
Maybe remove trailing whitespaces?

Reply:
@hyperxpro I do not see trailing whitespaces, also we have checks for that and I would expect that they will fail.
https://github.com/reactor/reactor-netty/blob/a86addb52836342b807e09345d9fcc99c4e0abb5/codequality/checkstyle.xml#L23-L27

Reply:
So these whitespaces between `volatile` to `Subscription`, and `Subscription` to `s` are normal as per code style?

Reply:
the spaces between volatile and Subscription not really. the second set of spaces is usual, to align the volatile variable with its associated `Atomic*FieldUpdater`

Reply:
Alright, that makes sense. I just started reading the source codes so not familiar with the coding style. Thanks for the input @violetagg @simonbasle :)